src/Analyzer/FileVisitor.php

```diff
@@ -133,6 +127,12 @@ public function enterNode(Node $node): void
             );
         }
 
+        if (null !== $this->classDescriptionBuilder && null !== $node->getDocComment()) {
```

Comment:
do we need the null check on `$this->classDescriptionBuilder`? I find it a bit confusing, as we are already using `$this->classDescriptionBuilder` a few lines above without any check

Reply:
I have got different errors without this check

Reply:
I am not sure I got what you mean but just out of curiosity: doesn't the check on line 28 prevent from having a null in `$this->classDescriptionBuilder` here?

And just to be super clear: all good from me, feel free to merge it

Reply:
The check on line 28 is inside an if that finish before the new check, this check is more generic

Reply:
split the if @fain182 ? What is confusing in this part?